client/src/components/BookDetail.test.js

```diff
@@ -0,0 +1,34 @@
+import BookDetail from './BookDetail'
+import { render } from '@testing-library/react'
+
+describe('BookDetail', () => {
+  it('renders correctly', () => {
+    const { container } = render(
+      <BookDetail
+        detailBook={[
+          {
+            volumeInfo: {
+              authors: [`Paul Tobin`],
+              imageLinks: {
+                thumbnail:
+                  'http://books.google.com/books/content?id=6D13DwAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api',
+              },
+              title: 'The Witcher, Band 2 - Fuchskinder',
+              categories: 'Fiction',
+              pageCount: '564',
+              printType: 'Book',
+              publishedDate: '12.12.1992',
+              publisher: 'Hüpfer Verlag',
+              language: 'DE',
+              description:
+                'Der Hexer ist wieder auf einer Spannenden Reise, auf der er Monster besiegt, Prinzessin Ciri rettet, mit Dryaden spricht und flotte Sprüche abgibt.',
+              subtitle: 'Der Witcher',
+            },
+          },
+        ]}
+        buttonName="Buch hinzufügen."
+      />
+    )
+    expect(container.firstChild).toMatchSnapshot()
+  })
+})
```

Comment:
If it helps to make the test a bit clearer, you could define all props above ``descripe`` and spread them into your component. 
```
const testProps = {
  formInputs: {
    location: '',
    date: '',
    players: '',
    winner: '',
    shots: '',
  },
  isSaveButtonShown: false,
  isEditFormShown: false,
  updateDirtyInputs: jest.fn(),
  handleChange: jest.fn(),
  showErrorMessage: jest.fn(),
  handleSubmit: jest.fn(),
  handleCancelEditModus: jest.fn(),
}

```

And then: 
```const { getByLabelText, getByTestId} = render(<CreateForm {...testProps} />) ```